src/VisualStudio/Xaml/Impl/Implementation/XamlOleCommandTarget.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    /// <summary>
+    /// This command target routes commands in .xaml files.
+    /// </summary>
+    internal sealed class XamlOleCommandTarget : AbstractOleCommandTarget
```

Comment:
However, It's going to introduce a significant amount of command handling to Xaml scenarios, and I worry about new things becoming visible/enabled because our `GetCommandState` implementations unconditionally enable them.